src/validation.cpp

```diff
@@ -144,6 +144,12 @@ class CChainState {
       */
     std::set<CBlockIndex*> g_failed_blocks;
 
+    /**
+     * the ChainState CriticalSection
+     * A lock that must be held when modifying this ChainState - held in ActivateBestChain()
```

Comment:
> must be held when modifying this ChainState

Seems like this statement is too broad; many aspects of this object are modified without this lock held (e.g. `g_failed_blocks`, `setBlockIndexCandidates`).

Reply:
I agree -- this is more of a lock on the logic in ABC than any particular data structure.  The comment in ABC itself is pretty good though.